connectors/spark/src/main/scala/io/unitycatalog/spark/auth/GcsVendedTokenProvider.java

```diff
@@ -0,0 +1,83 @@
+package io.unitycatalog.spark.auth;
+
+import io.unitycatalog.spark.UCHadoopConf;
+import org.apache.hadoop.conf.Configuration;
+import org.sparkproject.guava.base.Preconditions;
+import io.unitycatalog.client.model.GcpOauthToken;
+import com.google.cloud.hadoop.util.AccessTokenProvider;
+import com.google.cloud.hadoop.util.AccessTokenProvider.AccessToken;
+
+import java.io.IOException;
+import java.time.Instant;
+
+public class GcsVendedTokenProvider extends GenericCredentialProvider
+    implements AccessTokenProvider {
+
+  public static final String ACCESS_TOKEN_KEY = "fs.gs.auth.access.token.credential";
+  public static final String ACCESS_TOKEN_EXPIRATION_KEY = "fs.gs.auth.access.token.expiration";
+
+  private Configuration conf;
+
+  public GcsVendedTokenProvider() {
+  }
+
+  public GcsVendedTokenProvider(Configuration conf) {
```

Comment:
Is it required or not ?  if this constructor is required, please add coment why... seems like no usage here.